design/ceph/object/swift-and-keystone-integration.md

```diff
@@ -0,0 +1,312 @@
+---
+title: swift-and-keystone
+target-version: release-X.X
+---
+
+# Swift and Keystone Integration
+
+## Summary
+
+### Goals
+
+The goal of this proposal is to allow configuring the Swift API and
+the Keystone integration of Ceph RGW natively via the Object Store
+CRD, which allows native integration of the Rook operated Ceph RGW into
+OpenStack Clouds.
+
+Both changes are bundled together as one proposal because they will
+typically deployed together. It is unlikely to encounter a Swift
+object store API outside an OpenStack cloud and in the context of an
+OpenStack cloud the Swift API needs to be integrated with the Keystone
+authentication service.
+
+It must be possible to serve S3 and Swift for the same object store
+pool, e.g. by serving S3 and Swift from the same RGW instance.
+
+Any changes to the CRD must be future safe and cleanly allow extension
+to further technologies (such as LDAP authentication).
+
+### Non-Goals
+
+* Support for Keystone API versions below v3. API version v2 has long
+  been deprecated and [has been removed in the "queens" version of
+  Keystone](https://docs.openstack.org/keystone/xena/contributor/http-api.html)
+  which was released in 2018 and is now in extended maintenance mode
+  (which means it gets no more points releases and only sporadic bug
+  fixes/security fixes).
+
+* Authenticating Ceph RGW to Keystone via admin token – Only
+  authentication via an OpenStack service account will be supported.
```

Comment:
It's more clear IMO if you only mention what won't be supported in this section. If goals and non-goals are mixed in the same section it is confusing. 
```suggestion
```

Reply:
This section has been clarified.